### PR TITLE
fix: add min level to logger

### PR DESCRIPTION
**Pull request by sutterj**
# Pull Request

## Proposed Changes

- Adds minimum level to logger to hide debug logs in production
- Removes string interpolation from some logs for easier Splunk capture
- Adds mirror created log for Splunk

### Author/Contributor

- [ ] If documentation is needed for this change, has that been included in this pull request
- [X] run `npm run lint` and fix any linting issues that have been introduced
- [X] run `npm run test` and run tests
- [ ] If publishing new data to the public (scorecards, security scan results, code quality results, live dashboards, etc.), please request review from `@jeffrey-luszcz`

### Reviewer

- [X] Label as either `bug`, `documentation`, `enhancement`, `infrastructure`, `maintenance`, or `breaking`

### CODE REVIEW

File: src/utils/logger.ts [hunk 1]
```diff
@@ -13,20 +13,33 @@ const stringify = (obj: any) => {
...
-const getLoggerType = () => {
+const getLoggerConfig = (): {
+  type: 'pretty' | 'json'
+  minLevel: number
+} => {
   if (process.env.NODE_ENV === 'development') {
-    return 'pretty'
+    return {
+      type: 'pretty',
+      minLevel: 0,
```
Comment (ajhenry) [inline src/utils/logger.ts:23]: We have a logging level in the .env file too, we should use that

File: src/app/api/auth/lib/nextauth-options.ts [hunk 2]
```diff
@@ -157,18 +157,23 @@ export const nextAuthOptions: AuthOptions = {
           'No allowed handles or orgs specified, allowing all users.',
         )
 
+        authLogger.info('User is allowed to sign in:', profile.login)
```
Comment (ajhenry) [inline src/app/api/auth/lib/nextauth-options.ts:160]: iirc the second argument has to be an object

```suggestion
        authLogger.info('User is allowed to sign in:', { profile.login })
```

File: src/utils/logger.ts (hunk 1 above)
Comment (ajhenry) [reply src/utils/logger.ts:23]: https://github.com/github-community-projects/private-mirrors/blob/cdb56dad989e749f800b55e3ed7b93b38c1fafbc/env.mjs#L20

File: src/app/api/auth/lib/nextauth-options.ts (hunk 2 above)
Comment (sutterj) [reply src/app/api/auth/lib/nextauth-options.ts:160]: Logger puts string into data field directly in splunk, so change is not needed.

**Review by ajhenry — APPROVED** [review]
lgtm

**Pull request merged**